core/src/test/scala/sttp/openai/requests/completions/CompletionsDataSpec.scala

```diff
@@ -0,0 +1,132 @@
+    import sttp.openai.requests.completions.CompletionsRequestBody.CompletionsBody._
+
+    // given
+    val givenRequest = CompletionsRequestBody.CompletionsBody(
+      model = "text-davinci-003",
+      prompt = Some(SinglePrompt("Say this is a test")),
+      maxTokens = Some(7),
+      temperature = Some(0),
+      topP = Some(1),
+      n = Some(1),
+      stream = Some(false),
+      logprobs = None,
+      stop = Some(SingleStop("\n"))
+    )
+    val jsonRequest = fixtures.CompletionsFixture.jsonSinglePromptRequest
+
+    // when
+    val serializedJson = SnakePickle.writeJs(givenRequest)
+
+    // then
+    serializedJson.toString().filterNot(_.isWhitespace) shouldBe jsonRequest.filterNot(_.isWhitespace)
+  }
+
+  "Given completions of MultiplePrompt response as Json" should "be properly deserialized to case class" in {
+    import sttp.openai.requests.completions.CompletionsResponseData._
+    import sttp.openai.requests.completions.CompletionsResponseData.CompletionsResponse._
+
+    // given
+    val jsonResponse = fixtures.CompletionsFixture.jsonMultiplePromptResponse
+    val expectedResponse: CompletionsResponse = CompletionsResponse(
+      id = "cmpl-76D8UlnqOEkhVXu29nY7UPZFDTTlP",
+      `object` = "text_completion",
+      created = 1681714818,
+      model = "text-davinci-003",
+      choices = Seq(
+        Choices(
+          text = "\n\nThis is indeed a test",
+          index = 0,
+          logprobs = None,
+          finishReason = "length"
+        ),
+        Choices(
+          text = "\n\nYes, this is also",
+          index = 1,
+          logprobs = None,
+          finishReason = "length"
+        )
+      ),
+      usage = Usage(
+        promptTokens = 11,
+        completionTokens = 14,
+        totalTokens = 25
+      )
+    )
+
+    // when
+    val givenResponse: Either[Exception, CompletionsResponse] = SttpUpickleApiExtension.deserializeJsonSnake.apply(jsonResponse)
+
+    // then
+    givenResponse.value shouldBe expectedResponse
+  }
+
+  "Given completions of MultiplePrompt request as case class" should "be properly serialized to Json" in {
+    import sttp.openai.requests.completions.CompletionsRequestBody._
+    import sttp.openai.requests.completions.CompletionsRequestBody.CompletionsBody._
+
+    // given
+    val givenRequest = CompletionsRequestBody.CompletionsBody(
+      model = "text-davinci-003",
+      prompt = Some(MultiplePrompt(Seq("Say this is a test", "Say this is also a test"))),
+      maxTokens = Some(7),
+      temperature = Some(0),
+      topP = Some(1),
+      n = Some(1),
+      stream = Some(false),
+      logprobs = None,
+      stop = Some(SingleStop("\n"))
+    )
+    val jsonRequest = fixtures.CompletionsFixture.jsonMultiplePromptRequest
+
+    // when
+    val serializedJson = SnakePickle.writeJs(givenRequest)
+
+    // then
+    serializedJson.toString().filterNot(_.isWhitespace) shouldBe jsonRequest.filterNot(_.isWhitespace)
```

Comment:
**suggestion:** ‏
Instead of comparing string values we could map jsonRequest to `ujson.Value` e.g.
```suggestion
    val jsonRequest: ujson.Value = ujson.read(fixtures.CompletionsFixture.jsonMultiplePromptRequest)

    // when
    val serializedJson: json.Value = SnakePickle.writeJs(givenRequest)

    // then
    serializedJson shouldBe jsonRequest
```